src/Uno.UI.Runtime.Skia.Win32/ApplicationMode/DataTransfer/DragDrop/Win32DragDropExtension.DropTarget.cs

```diff
@@ -0,0 +1,160 @@
+		hResult = enumFormatEtc->Next(formatBufferLength, formatBuffer, &fetchedFormatCount);
+		if (hResult.Failed)
+		{
+			this.LogError()?.Error($"{nameof(PInvoke.RegisterDragDrop)} failed: {Win32Helper.GetErrorMessage(hResult)}");
+			return HRESULT.E_UNEXPECTED;
+		}
+
+		var position = new System.Drawing.Point(pt.x, pt.y);
+
+		var success = PInvoke.ScreenToClient(_hwnd, ref position);
+		if (!success) { this.LogError()?.Error($"{nameof(PInvoke.ScreenToClient)} failed: {Win32Helper.GetErrorMessage()}"); }
+		var scaledPosition = GetScaledPosition(position.X, position.Y);
+
+		var src = new DragEventSource(scaledPosition, grfKeyState);
+
+		var formats = new Span<FORMATETC>(formatBuffer, (int)fetchedFormatCount);
+		if (this.Log().IsEnabled(LogLevel.Trace))
+		{
+			var log = $"{nameof(IDropTarget.Interface.DragEnter)} @ {position}, formats: ";
+			foreach (var format in formats)
+			{
+				log += (CLIPBOARD_FORMAT)format.cfFormat + " ";
+			}
+			this.Log().Trace(log);
+		}
+
+		var package = new DataPackage();
+
+		var formatEtcList = formats.ToArray();
+		var formatList =
+			formatEtcList
+			.Where(static formatetc =>
+			{
+				if (!Enum.IsDefined((CLIPBOARD_FORMAT)formatetc.cfFormat))
+				{
+					return false;
+				}
+				if (formatetc.tymed != (uint)TYMED.TYMED_HGLOBAL)
+				{
+					typeof(Win32DragDropExtension).LogError()?.Error($"{nameof(IDropTarget.Interface.DragEnter)} found {Enum.GetName((CLIPBOARD_FORMAT)formatetc.cfFormat)}, but {nameof(TYMED)} is not {nameof(TYMED.TYMED_HGLOBAL)}");
+					return false;
+				}
+
+				return true;
+			})
+			.Select(f => (CLIPBOARD_FORMAT)f.cfFormat)
+			.ToList();
+
+		var mediumsToDispose = new List<STGMEDIUM>();
+		using var mediumsDisposable = new DisposableStruct<List<STGMEDIUM>>(static list =>
+		{
+			foreach (var medium in list)
+			{
+				PInvoke.ReleaseStgMedium(&medium);
+			}
+		}, mediumsToDispose);
+		Win32ClipboardExtension.ReadContentIntoPackage(package, formatList, format =>
+		{
+			var formatEtc = formatEtcList.First(f => f.cfFormat == (int)format);
+			dataObject->GetData(formatEtc, out STGMEDIUM medium);
+			mediumsToDispose.Add(medium);
+			return medium.u.hGlobal;
+		});
+
+		// Create DragUI for visual feedback during drag operation
+		var dragUI = CreateDragUIForExternalDrag(dataObject, formatEtcList);
+
+		// DROPEFFECT and DataPackageOperation have the same binary representation
+		var info = new CoreDragInfo(src, package.GetView(), (DataPackageOperation)(*pdwEffect), dragUI);
+		_coreDragDropManager.DragStarted(info);
+
+		*pdwEffect = (DROPEFFECT)_manager.ProcessMoved(src);
+
+		return HRESULT.S_OK;
+	}
+
+	unsafe HRESULT IDropTarget.Interface.DragOver(MODIFIERKEYS_FLAGS grfKeyState, POINTL pt, DROPEFFECT* pdwEffect)
+	{
+		var position = new System.Drawing.Point(pt.x, pt.y);
+		var success = PInvoke.ScreenToClient(_hwnd, ref position);
+		if (!success) { this.LogError()?.Error($"{nameof(PInvoke.ScreenToClient)} failed: {Win32Helper.GetErrorMessage()}"); }
```

Comment:
Same here, what can be done about it? What's the effect of a failure?

Reply:
This should not be happening unless something goes wrong in Windows